bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigDescriptionRegistry.java

```diff
@@ -165,4 +177,102 @@ public ConfigDescription getConfigDescription(URI uri) {
         return getConfigDescription(uri, null);
     }
 
+    /**
+     * Returns a config description for a given URI. The URI is assumed to be a concrete implementation.
+     * <p>
+     * If more than one {@link ConfigDescriptionProvider} is registered for the requested URI, then the returned
+     * {@link ConfigDescription} will contain the data from all providers.
+     * <p>
+     * No checking is performed to ensure that multiple providers don't provide the same configuration data. It is up to
+     * the binding to ensure that multiple sources (eg static XML and dynamic binding data) do not contain overlapping
+     * information.
+     *
+     * @param uri
```

Comment:
Agreed - this PR needs a tidy (and tests). It’s provided early for comment to see if it even fits the concept as I don’t want to waste time if it’s not something that’s going to be merged.